src/features/references/editor/ReferenceDetailsCard.tsx

```diff
@@ -0,0 +1,267 @@
+    return '[' + content.toString() + ']';
+  } else {
+    return content.toString();
+  }
+};
+
+const DataTextInput = ({
+  content,
+  id,
+  onChangeHandler,
+}: {
+  content: string;
+  id: string;
+  onChangeHandler: CallableFunction;
+}) => {
+  const [value, setValue] = useState(content);
+
+  return (
+    <input
+      className="w-full border bg-slate-50 px-2 py-0.5"
+      name={id}
+      title={id} // required for unit testing to find this element
+      type="text"
+      value={value}
+      onBlur={(evt) => {
+        onChangeHandler(id, evt.target.value);
+      }}
+      onChange={(evt) => {
+        setValue(evt.target.value);
+      }}
+    />
+  );
+};
+
+const TableDataCell = ({
+  contentData,
+  id,
+  onChangeHandler,
+  editable,
+  editing,
+}: {
+  contentData: string | Author[];
+  id: string;
+  onChangeHandler: CallableFunction;
+  editable: boolean;
+  editing: boolean;
+}) => {
+  if (!editing || !editable || typeof contentData != 'string') {
+    return <UnEditableTableDataCell contentData={contentData} id={id} />;
+  } else {
+    return <EditableTableDataCell contentData={contentData} id={id} onChangeHandler={onChangeHandler} />;
+  }
+};
+
+const EditableTableDataCell = ({
+  contentData,
+  id,
+  onChangeHandler,
+}: {
+  contentData: string;
+  id: string;
+  onChangeHandler: CallableFunction;
+}) => (
+  <td className="w-auto p-5">
+    <DataTextInput content={contentData} id={id} onChangeHandler={onChangeHandler} />
+  </td>
+);
+
+const UnEditableTableDataCell = ({ contentData, id }: { contentData: string | Author[]; id: string }) => {
+  const contentString = getContentString(id, contentData);
+
+  return (
+    <td className="w-auto p-5">
+      <span className="leading-[30px]">{contentString}</span>
+    </td>
+  );
+};
+
+const TableHeadCell = ({
+  content,
+  colSpan,
+  header,
+  id,
+  editing,
+  setEditing,
+}: {
+  content: string;
+  colSpan?: number;
+  header?: boolean;
+  id: string;
+  editing?: boolean;
+  setEditing?: CallableFunction;
+}) => {
+  const colSpanString = { ['colSpan']: colSpan };
+
+  return (
+    <th
+      className={cx(
+        { 'rounded-t bg-card-bg-header p-3 text-white': header },
+        { 'w-1/6 p-6': !header },
+        'text-sm uppercase',
+      )}
+      key={content}
+      {...colSpanString}
+    >
+      <label htmlFor={id}>{content}</label>
+      {IconButtons(header, editing, setEditing)}
+    </th>
+  );
+};
+
+const IconButtons = (header?: boolean, editing?: boolean, setEditing?: CallableFunction): ReactElement[] => {
+  if (!header || editing === undefined || setEditing === undefined) {
+    return [];
+  } else {
+    return [
+      IconButton({ hiddenWhenEditing: true, icon: EditIcon(), title: 'Edit Reference', editing, setEditing }),
+      IconButton({
+        hiddenWhenEditing: false,
+        icon: CloseIcon(),
+        title: 'Finished Editing Reference',
+        editing,
+        setEditing,
+      }),
+    ];
+  }
+};
```

Comment:
You should move this logic in `TableHeadCell`, like this:
```suggestion
    {header && setEditing && (
      editing ?
        <IconButton editing={false} icon={<CloseIcon />} setEditing={setEditing} title="Finished Editing Reference" /> :
        <IconButton editing={true} icon={<EditIcon />} setEditing={setEditing} title="Edit Reference" />
    )}
  </th>
);
```